src/aids-env.adb

```diff
@@ -1,58 +1,85 @@
 with Ada.Text_IO; use Ada.Text_IO;
+with Ada.Strings.Fixed;
 
 package body Aids.Env is
+    Function Find(Env   : in Typ; 
+                  Key   : in String;
+                  Found : in out Boolean
+                 ) return String is
+    Begin
+        Found := Env.Contains( Key );
+        Return (if not Found then "" else Env(Key));
+    End Find;
+
     function Index_Of(Line: in String; X: in Character; Result: out Integer) return Boolean is
+        Index : Natural renames Ada.strings.Fixed.Index(
+           Source  => Line,      -- The String we're searching,
+           Pattern => (1 => X),  -- the *Substring* to search for;
+           From    => Line'First -- the point we start from.
+          );
     begin
-        for Index in Line'Range loop
-            if Line(Index) = X then
-                Result := Index;
-                return True;
-            end if;
-        end loop;
-        return False;
+        Return Found : constant Boolean := Index in Positive do
+            Result:= Index;
+        End return;
     end;
 
     function Slurp(File_Path: String) return Env_Hashed_Map.Map is
-        Result: Env_Hashed_Map.Map;
         File: File_Type;
         Line_Number: Integer := 1;
     begin
-        Open(File => File,
-             Mode => In_File,
-             Name => File_Path);
-        while not End_Of_File(File) loop
-            declare
-                Line: String := Get_Line(File);
-                Index : Integer;
-            begin
-                if Index_Of(Line, '=', Index) then
-                    declare
-                        Key : Unbounded_String := To_Unbounded_String(Line(Line'First..Index-1));
-                        Value : Unbounded_String := To_Unbounded_String(Line(Index+1..Line'Last));
-                    begin
-                        Env_Hashed_Map.Insert(Result, Key, Value);
-                    end;
-                else
-                    raise Syntax_Error with (File_Path 
-                                             & ":" 
-                                             & Integer'Image(Line_Number)
-                                             & ": Expected separator `=`");
-                end if;
+        Return Result: Env_Hashed_Map.Map do
+            Open(File => File,
+                 Mode => In_File,
+                 Name => File_Path);
+            while not End_Of_File(File) loop
+                declare
+                    Line: String := Get_Line(File);
+                    Index : Integer;
+                begin
+                    if Index_Of(Line, '=', Index) then
+                        declare
+                            Use Ada.Strings;
+                            Prev  : Natural  renames Positive'Pred( Index );
```

Comment:
why use `renames` instead of `:=` is the behavior different?

Reply:
There are some differences as you can `renames` other things like `package` and `function` and record-components; this form is the renaming of a function/attribute result and is equivalent to `Prev  : Constant Natural  := Positive'Pred( Index );`.

I do prefer the `renames` form as it does indicate that I'm just "_using some name_" for the thing.